File: sample_java/src/main/java/com/charlezz/sample_java/MainActivity.java
package com.charlezz.sample_java;

import android.os.Bundle;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.PagerSnapHelper;
import androidx.recyclerview.widget.RecyclerView;

import com.charlezz.sample_java.databinding.ActivityMainBinding;

import life.sabujak.pickle.Pickle;
import life.sabujak.pickle.data.entity.PickleError;
import life.sabujak.pickle.ui.dialog.OnResultListener;
import life.sabujak.pickle.ui.dialog.PickleResult;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;
    private String[] items = new String[]{"Basic","Insta"};
    private ImageAdapter adapter = new ImageAdapter();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        this.binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        binding.spinner.setAdapter(new ArrayAdapter<>(this,android.R.layout.simple_list_item_1, items));
        binding.load.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                switch (items[binding.spinner.getSelectedItemPosition()]){
                    case "Basic":
                        Pickle.start(getSupportFragmentManager(), new OnResultListener() {
                            @Override
                            public void onSuccess(PickleResult result) {
                                adapter.submitList(result.getMediaList());
                            }

                            @Override
                            public void onError(PickleError error) {
                                Toast.makeText(MainActivity.this, error.getErrorCode().name(), Toast.LENGTH_SHORT).show();
                            }
                        });
                        break;
                    case "Insta":
                        break;
                }
            }
        });

        binding.recyclerView.setLayoutManager(new LinearLayoutManager(this, RecyclerView.HORIZONTAL, false));
        binding.recyclerView.setAdapter(adapter);
        new PagerSnapHelper().attachToRecyclerView(binding.recyclerView);

    }
}

File: sample/src/main/java/life/sabujak/picklesample/MainActivity.kt
package life.sabujak.picklesample

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import life.sabujak.pickle.Config
import life.sabujak.pickle.Pickle
import life.sabujak.pickle.data.entity.PickleResult
import life.sabujak.pickle.ui.common.OnResultListener
import life.sabujak.picklesample.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding: ActivityMainBinding =
            DataBindingUtil.setContentView(this, R.layout.activity_main)
        binding.basic.setOnClickListener {
            Pickle.start(supportFragmentManager, object : OnResultListener {
                override fun onSuccess(result: PickleResult) {
                    result.mediaList.forEach {
                        Log.e("MainActivity", it.toString())
                    }
                }

                override fun onFailure(t: Throwable) {
                }
            })
        }

        binding.insta.setOnClickListener{
            Pickle.start(supportFragmentManager)
        }

    }
}
